internal/exec/shell_utils.go

```diff
@@ -185,13 +185,24 @@ func execTerraformShellCommand(
 		}
 	}()
 
-	// Set the Terraform environment variables to reference the var file
-	componentEnvList = append(componentEnvList, fmt.Sprintf("TF_CLI_ARGS_plan=-var-file=%s", varFile))
-	componentEnvList = append(componentEnvList, fmt.Sprintf("TF_CLI_ARGS_apply=-var-file=%s", varFile))
-	componentEnvList = append(componentEnvList, fmt.Sprintf("TF_CLI_ARGS_refresh=-var-file=%s", varFile))
-	componentEnvList = append(componentEnvList, fmt.Sprintf("TF_CLI_ARGS_import=-var-file=%s", varFile))
-	componentEnvList = append(componentEnvList, fmt.Sprintf("TF_CLI_ARGS_destroy=-var-file=%s", varFile))
-	componentEnvList = append(componentEnvList, fmt.Sprintf("TF_CLI_ARGS_console=-var-file=%s", varFile))
+	// Define the Terraform commands that may use var-file configuration
+	tfCommands := []string{"plan", "apply", "refresh", "import", "destroy", "console"}
+
+	// Check for existing var-file arguments in TF_CLI environment variables
+	for _, cmd := range tfCommands {
+		envVar := fmt.Sprintf("TF_CLI_ARGS_%s", cmd)
+		existing := os.Getenv(envVar)
+		if existing != "" {
+			u.LogWarning(atmosConfig, fmt.Sprintf("Found Terraform environment variable '%s' which may conflict with or be overwritten by Atmos", envVar))
```

Comment:
```suggestion
			u.LogWarning(atmosConfig, fmt.Sprintf("Warning: Detected '%s' set in the environment; this may interfere with Atmos's control of Terraform.", envVar))
```